bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/internal/items/ItemBuilderTest.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.internal.items;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.eclipse.smarthome.core.items.ActiveItem;
```

Comment:
Since `ActiveItem` is not public API and you are introducing the builder pattern to create items, shall we move this class to an `internal` package and remove the `@Deprecated` annotation on it?

Reply:
No!
It currently _is_ public API. And moving it into an internal package will not only be "API"-breaking, but will also render all concrete item implementations non-usable, as this interface defines the public methods on them. 
The current Item-API design is flawed. There is nothing we can do about that, apart from rewriting major parts of it. But that'll come at a high price on consumers' side.